### PR TITLE
Introduce black, flake8, isort

### LINKED ISSUE #25: Introduce auto code formatting
See https://github.com/treasure-data/pytd/pull/24#discussion_r299321461

### PR DESCRIPTION
**Pull request by chezou**
This PR introduces
- black
- isort
- flake8

Also added pre-commit configuration so that pre-commit hook will format automatically.

close #25

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,13 +18,14 @@ before_install:
 
   - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION nose mock
```
Comment (takuti) [inline .travis.yml]: You can delete `nose`

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
```
Comment (takuti) [inline .pre-commit-config.yaml]: Fix indentations.

File: setup.py [hunk 3]
```diff
@@ -39,37 +41,40 @@ def setup_package():
...
+            "pandas>=0.22.0",
+            "td-client>=0.12.0",
+            "pytz>=2018.5",
         ],
         extras_require={
-            'spark': ['pyspark>=2.4.0', 'pyarrow>=0.11.0'],
-        })
+            "spark": ["pyspark>=2.4.0", "pyarrow>=0.11.0"],
+            "dev": ["black", "isort", "flake8"],
+        },
```
Comment (takuti) [inline setup.py]: We can create `test` dependency with `["pytest"]` and modify `noxfile.py` as:

```py
@nox.session
def tests(session):
    session.install(".[dev,test,spark]")
    session.run("pytest", "-v")
```

File: .pre-commit-config.yaml (hunk 2 above)
Comment (chezou) [reply .pre-commit-config.yaml]: Ah, this was generated pre-commit, but will align with other stuff

File: .travis.yml (hunk 1 above)
Comment (chezou) [reply .travis.yml]: will remove `mock` as well

File: setup.py (hunk 3 above)
Comment (chezou) [reply setup.py]: fixed 4f3f37c

File: .pre-commit-config.yaml (hunk 2 above)
Comment (chezou) [reply .pre-commit-config.yaml]: fixed 4f3f37c

File: .travis.yml (hunk 1 above)
Comment (chezou) [reply .travis.yml]: removed 4f3f37c

File: .gitignore [hunk 4]
```diff
@@ -1,7 +1,126 @@
...
+coverage.xml
+*.cover
+.hypothesis/
+.pytest_cache/
+
+# Translations
+*.mo
+*.pot
+
+# Django stuff:
```
[the remark this answers is not in the document]
Comment (chezou) [reply .gitignore]: removed 4f3f37c

File: setup.py [hunk 5]
```diff
@@ -39,37 +41,41 @@ def setup_package():
...
+            "presto-python-client>=0.6.0",
+            "pandas>=0.22.0",
+            "td-client>=0.12.0",
+            "pytz>=2018.5",
         ],
         extras_require={
-            'spark': ['pyspark>=2.4.0', 'pyarrow>=0.11.0'],
-        })
+            "spark": ["pyspark>=2.4.0", "pyarrow>=0.11.0"],
+            "dev": ["black", "isort", "flake8"],
```
Comment (takuti) [inline setup.py]: Just noticed a bit of a contradiction - while `pytd` officially supports Python 3.5+, one extra dependency `black` requires Python 3.6+.

Can you delete `dev` extra require section from `setup.py` for now? 

Keep using `black`, `isort` and `flake8`, but they should be explicitly installed via `pip` with Python 3.6+. Please update `noxfile.py` and README accordingly with this notice.

File: README.md [hunk 6]
```diff
@@ -173,3 +173,23 @@ con = td.connect(apikey=apikey, endpoint=endpoint, writer=writer)
...
 td.to_td(df, 'mydb.buzz', con, if_exists='replace', index=False)
 ```
+
+## For developers
+
+We use [black](https://black.readthedocs.io/en/stable/) and [isort](https://github.com/timothycrosley/isort) as a formatter, and [flake8](http://flake8.pycqa.org/en/latest/) as a linter. Our CI checks format with them.
+
+We highly recommend you to introduce [pre-commit](https://pre-commit.com/) to ensure your commit follows required format.
+
+You can install pre-commit as follows:
```
Comment (takuti) [inline README.md]: Personally I would also like to mention how to skip pre-commit check here: `git commit --no-verify`

[Recent updates on pandas contributing guide](https://github.com/pandas-dev/pandas/pull/27233) could be a good reference.

Comment (chezou) [reply README.md]: thanks for the reference. I will add to skip pre-commit check.

File: setup.py (hunk 5 above)
Comment (chezou) [reply setup.py]: Good catch. Removed `dev` extras and add a note on README. 76a030d

File: appveyor/requirements.txt [hunk 7]
```diff
@@ -3,7 +3,8 @@ pandas
 td-client
 pyspark
 pyarrow
-nose
+pytest
 mock
```
Comment (takuti) [inline appveyor/requirements.txt]: We can delete `mock` from AppVeyor requirements as well.

Comment (chezou) [reply appveyor/requirements.txt]: good catch. removed eb13381

**Comment by chezou** [comment]
CI statuses seem still pending, but Travis status page shows CI was succeeded. I'll merge it. Thanks for your review, takuti!

**Pull request merged**